doc/developers/contributing.rst

```diff
@@ -790,17 +790,66 @@ In the following example, k is deprecated and renamed to n_clusters::
 
     import warnings
 
-    def example_function(n_clusters=8, k=None):
-        if k is not None:
+    def example_function(n_clusters=8, k='not_used'):
```

Comment:
We should wait and see if others think if this is a good idea. @jnothman @ogrisel @rth ?